src/util/JoinAlgorithms/JoinAlgorithms.h

```diff
@@ -1133,4 +1329,26 @@ void zipperJoinForBlocksWithoutUndef(LeftBlocks&& leftBlocks,
   impl.template runJoin<DoOptionalJoin>();
 }
 
+template <typename LeftBlocks, typename RightBlocks, typename LessThan,
+          typename LeftProjection = std::identity,
+          typename RightProjection = std::identity,
+          typename DoOptionalJoinTag = std::false_type>
+void zipperJoinForBlocksWithPotentialUndef(LeftBlocks&& leftBlocks,
```

Comment:
The comment and maybe the concepts should specify that this only works for single column join, because it assumes in all places that the undefined values form a contiguous range at the beginning.

Reply:
(also for the Impl).

Reply:
Which concepts do you have in mind?